src/tests/End2EndTestsMain.cpp

```diff
@@ -16,7 +16,28 @@
 #include "tests/WebnnTest.h"
 
 int main(int argc, char** argv) {
-    InitWebnnEnd2EndTestEnvironment();
+    std::string device = "Default";
+    for (int i = 1; i < argc; ++i) {
```

Comment:
Do you mean we need to run the end2end multi-times on different device?

Reply:
As discussed with @BruceDai , running on one device per time can help to collect and parse tests results, am I right? @BruceDai

Reply:
@mingmingtasd Yes, if we ran test per device(CPU/GPU/...), we could clearly get test result with specified device. If you supported running test with multi supported devices one time, please consider outputting distinguished test result by device which would help test tool to parse output result.